### PR TITLE
Define LinearRegressionRequest and Response messages

### PR DESCRIPTION
**Pull request by inf0rmatiker**
Title says it all

### CODE REVIEW

File: src/main/proto/sustain.proto [hunk 1]
```diff
@@ -64,11 +64,39 @@ message KMeansClusteringResponse {
...
+  double epsilon = 8;
+  double convergenceTolerance = 9;
+  double regularizationParam = 10;
+  bool setStandardization = 11;
+  bool fitIntercept = 12;
 }
 
 message LinearRegressionResponse {
-  // TODO - fill out
+  repeated Collection collections = 1;
```
Comment (hamersaw) [inline src/main/proto/sustain.proto:82]: I don't think we need this. If it's included on the Request than the client-side already has the information.

File: src/main/proto/sustain.proto [hunk 2]
```diff
@@ -64,11 +64,39 @@ message KMeansClusteringResponse {
 }
 
 message LinearRegressionRequest {
-  // TODO - fill out
+  repeated Collection collections = 1;
+  repeated string gisJoins = 2;
```
Comment (hamersaw) [inline src/main/proto/sustain.proto:68]: Should we define this on the ModelRequest? It seems to be a field that could be used in all model training.

File: src/main/proto/sustain.proto [hunk 3]
```diff
@@ -64,11 +64,39 @@ message KMeansClusteringResponse {
...
+  message LinearRegressionSummary {
+    string gisJoin = 1;
+    double totalIterations = 2;
+    double rmseResidual = 3;
+    double r2Residual = 4;
+    double intercept = 5;
+    repeated double slopeCoefficients = 6;
+    repeated double objectiveHistory = 7;
+  }
+  repeated LinearRegressionSummary summaries = 2;
```
Comment (hamersaw) [inline src/main/proto/sustain.proto:93]: The rpc definition for modelRequest is a stream. The LinearRegressionResponse could simply be the fields in the LinearRegressionSummary message as we will be returning multiple instances through rpc.

File: src/main/proto/sustain.proto (hunk 2 above)
Comment (inf0rmatiker) [reply src/main/proto/sustain.proto:68]: Agreed

**Comment by inf0rmatiker** [comment]
- Updated the proto definition of `LinearRegressionResponse` -- puts all fields at root level of object.
- Removed Collection from `LinearRegressionResponse` and moved it into `ModelRequest`, since it will be used by all model requests.

File: src/main/proto/sustain.proto [hunk 4]
```diff
@@ -64,11 +65,33 @@ message KMeansClusteringResponse {
...
+  double rmseResidual = 3;
+  double r2Residual = 4;
+  double intercept = 5;
+  repeated double slopeCoefficients = 6;
+  repeated double objectiveHistory = 7;
+}
+
+message Collection {
+  string name = 1;
+  string label = 2;
```
Comment (menuka94) [inline src/main/proto/sustain.proto:93]: What does `label` represent?

Comment (inf0rmatiker) [reply src/main/proto/sustain.proto:93]: This is the column, or feature we are trying to predict.

**Review by menuka94 — APPROVED** [review]

**Pull request merged**